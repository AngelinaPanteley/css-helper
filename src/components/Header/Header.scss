@import '../../variables';

.Header {
  @include flex-wrapper();
  
  padding: 0 40px;
  background: $color-gray;
}