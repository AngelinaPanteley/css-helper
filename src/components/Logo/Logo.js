import React from 'react';
import { Link } from 'react-router-dom';

import styles from './Logo.scss';
import logoImg from '../../assets/images/logo.png';

const logo = (props) => {
  return (
    <Link to='/' className={styles.Logo}>
      <img src={logoImg} alt="logo" className={styles.Logo_image} />
    </Link>
  )
}

export default logo;