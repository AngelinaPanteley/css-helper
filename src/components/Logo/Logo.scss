@import '../../variables';

.Logo {
  @include x-center;

  top: 0;

  &_image {
    width: 100px;
  }
}