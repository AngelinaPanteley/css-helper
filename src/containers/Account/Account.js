import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link, withRouter } from 'react-router-dom';

import styles from './Account.scss';

import Icon from '../../components/UI/Icon/Icon';
import Dropdown from '../../components/UI/Dropdown/Dropdown';
import Auxiliary from '../../hoc/Auxiliary/Auxiliary';
import * as actions from '../../store/actions/index';

class Account extends Component {
  state = {
    isDropdownOpen: false,
  }

  onAccountClick = (e) => {
    e.preventDefault();
    this.toggleDropdown();
  }

  toggleDropdown = () => {
    this.setState(prevState => {
      return {
        isDropdownOpen: !prevState.isDropdownOpen
      }
    });
  }

  render() {
    let links = [
      {
        title: 'Log In',
        route: '/login',
      },
      {
        title: 'Create Account',
        route: '/auth',
      }
    ];

    if (this.props.isAuth) {
      let showBy = localStorage.getItem('showBy');
      if (!showBy) {
        showBy = 5;
      }
      links = [
        {
          title: 'My Savings',
          route: `/savings/${showBy}/1`,
        },
        {
          title: 'Logout',
          route: '',
          click: this.props.onLogout,
        }
      ];
    }

    return (
      <Auxiliary>
        <div className={styles.Account}>
          {
            links.map((link) => {
              return (
                <span className={styles.Account_Link_Wrapper} key={link.route}>
                  <Auxiliary>
                    {
                      link.route
                        ?
                        <Link to={link.route}
                          className={styles.Account_Link}
                          onClick={link.click}>
                          {link.title}
                        </Link>
                        :
                        <a className={styles.Account_Link}
                          onClick={(e) => {
                            e.preventDefault();
                            link.click()
                          }}>
                          {link.title}
                        </a>
                    }
                  </Auxiliary>
                </span>
              )
            })
          }
        </div>
        <div className={styles.Account_Mobile}>
          <a href="" onClick={this.onAccountClick}>
            <Icon icon='account' />
          </a>
          {
            <Dropdown isOpen={this.state.isDropdownOpen}
              links={links} right
              onClose={this.toggleDropdown} />
          }
        </div>
      </Auxiliary>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    editorNames: state.editors.names,
    isAuth: !!state.auth.userId,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    onLogout: () => dispatch(actions.logout()),
  }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Account));