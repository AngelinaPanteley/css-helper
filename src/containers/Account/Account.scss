@import '../../variables';

.Account {
  @include mobile {
    display: none;
  }

  &_Mobile {
    display: none;

    @include mobile {
      display: block;
    }
  }

  &_Link {
    text-decoration: none;

    &_Wrapper {
      margin: 0 10px;

      &:last-child {
        margin-right: 0;
      }

      &:first-child {
        margin-left: 0;
      }
    }
  }
}