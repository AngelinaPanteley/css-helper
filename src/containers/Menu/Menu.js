import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import styles from './Menu.scss';
import * as actions from '../../store/actions/index';
import Icon from '../../components/UI/Icon/Icon';
import Dropdown from '../../components/UI/Dropdown/Dropdown';

class Menu extends Component {
  state = {
    isDropdownOpen: false,
  }

  onMenuClick = (e) => {
    e.preventDefault();
    this.toggleDropdown();
  }

  toggleDropdown = () => {
    this.setState(prevState => {
      return {
        isDropdownOpen: !prevState.isDropdownOpen
      }
    });
  }

  render() {
    const links = this.props.editorNames.map((elem) => {
      return {
        title: 'Create ' + elem[0].toUpperCase() + elem.slice(1),
        route: '/' + elem,
      }
    });

    return (
      <div className={styles.Menu}>
        <a href="" onClick={this.onMenuClick}>
          <Icon icon='menu' />
        </a>
        {
          <Dropdown isOpen={this.state.isDropdownOpen}
            links={links} left
            onClick={this.props.turnEditingModeOff}
            onClose={this.toggleDropdown} />
        }
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    editorNames: state.editors.names,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    turnEditingModeOff: () => dispatch(actions.turnEditingModeOff()),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Menu);