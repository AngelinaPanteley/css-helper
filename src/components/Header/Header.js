import React from 'react';

import styles from './Header.scss'

import Logo from '../Logo/Logo';
import Menu from '../../containers/Menu/Menu';
import Account from '../../containers/Account/Account';

const header = (props) => {
  return (
    <div className={styles.Header} >
      <Menu />
      <Logo />
      <Account />
    </div>
  );
}

export default header;