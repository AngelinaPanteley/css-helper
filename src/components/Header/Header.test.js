"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { expect as chaiExpect } from 'chai';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import { MemoryRouter } from "react-router-dom";
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import Header from './Header.js';
import Menu from '../../containers/Menu/Menu.js';
import Account from '../../containers/Account/Account.js';
import Logo from '../Logo/Logo.js';
import reducer from '../../store/reducers/index';

configure({ adapter: new Adapter() });

const store = createStore(reducer);

describe('>>>Header', () => {

  //SNAPSHOTS

  const headerComponent = <Provider store={store}><MemoryRouter>
    <Header />
  </MemoryRouter></Provider>;

  const component = renderer.create(headerComponent);

  it('renders control correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  //SHALLOW RENDERING

  let wrapper;
  const shallowComponent = <Header />;

  beforeEach(() => {
    wrapper = shallow(shallowComponent);
  });

  it('renders Menu', () => {
    chaiExpect(wrapper.find('div')).to.have.lengthOf(1);
  });
  it('renders Logo', () => {
    chaiExpect(wrapper.find(Logo)).to.have.lengthOf(1);
  });
  it('renders Account', () => {
    chaiExpect(wrapper.find(Account)).to.have.lengthOf(1);
  });
});
