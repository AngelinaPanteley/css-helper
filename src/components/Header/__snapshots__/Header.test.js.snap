// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Header renders control correctly 1`] = `
<div>
  <div>
    <a
      href=""
      onClick={[Function]}
    >
      <i
        className="undefined undefined"
      />
    </a>
    <div
      className="  "
    >
      <ul />
    </div>
  </div>
  <a
    href="/"
    onClick={[Function]}
  >
    <img
      alt="logo"
      src="logo.png"
    />
  </a>
  <div>
    <span>
      <a
        href="/login"
        onClick={[Function]}
      >
        Log In
      </a>
    </span>
    <span>
      <a
        href="/auth"
        onClick={[Function]}
      >
        Create Account
      </a>
    </span>
  </div>
  <div>
    <a
      href=""
      onClick={[Function]}
    >
      <i
        className="undefined undefined"
      />
    </a>
    <div
      className="  "
    >
      <ul>
        <li>
          <a
            href="/login"
            onClick={[Function]}
          >
            Log In
          </a>
        </li>
        <li>
          <a
            href="/auth"
            onClick={[Function]}
          >
            Create Account
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
